easybuild/easyblocks/e/elpa.py

```diff
@@ -159,6 +161,24 @@ def run_all_steps(self, *args, **kwargs):
             self.cfg.update('configopts', '--with-mpi=no')
             self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
 
+        # Add CUDA features
+        cuda_is_dep = 'CUDA' in [i['name'] for i in self.cfg.dependencies()]
```

Comment:
All this, including the "cuda" extra_var, can be simplified to
if get_software_root('CUDA'):
 ...

That way you can drop the elif statements and the cuda extra_var.

Reply:
That's actually not fully true...

If the `CUDA` module was loaded manually, then `$EBROOTCUDA` is also defined (and that's all that `get_software_root` does: it grabs the corresponding `$EBROOT*` environment variable).

So this is actually a better approach...

We should have proper support for this in framework though, so you can just do `self.cfg.is_dep('CUDA')`

Reply:
For JSC, CUDA is a dep of MPI by default (and that MPI works regardless of whether or not there actually is a GPU through the use of UCX variables). So `get_software_root('CUDA')` is not enough, you need to know if it is actually a dep to decide whether to trigger a GPU build.

Reply:
@ocaisa if CUDA is a dep of MPI regardless of whether or not there actually is a GPU, don’t you want to trigger a GPU build also regardless of whether or not there actually is a GPU?

Reply:
@smoors No, because then that GPU-enabled build will (typically) not work on a system without a GPU (whereas OpenMPI will work just fine on that system as long as you set the appropriate btl). So at JSC, the GPU supported build is usually explicit